### PR TITLE
[corlib] Use ArrayPool from coreclr

### PR DESCRIPTION
**Pull request by marek-safar**

### CODE REVIEW

File: mcs/class/corlib/corert/RuntimeThread.cs [hunk 1]
```diff
@@ -33,5 +33,11 @@ public static bool SpinWait (int iterations)
 			Thread.SpinWait (iterations);
 			return true;
 		}
+
+		public static int GetCurrentProcessorId ()
+		{
+			// TODO: Implement correctly
+			return 1;
```
Comment (akoeplinger) [inline mcs/class/corlib/corert/RuntimeThread.cs]: should we use `Environment.CurrentManagedThreadId` instead? That's what they're doing in some cases in CoreCLR: https://github.com/dotnet/coreclr/blob/29d5dc25ab6fd078cc349e7f7e21aaeacda59acc/src/mscorlib/src/Internal/Runtime/Augments/RuntimeThread.cs#L234-L238

**Comment by luhenry** [comment]
@mareks could you please rebase on master?

**Comment by luhenry** [comment]
@marek-safar @akoeplinger there is a new ArrayPool<T> type in net_4_x, monodroid and monotouch profiles, is it acceptable?

**Comment by luhenry** [comment]
@monojenkins commit csproj

**Comment by marek-safar** [comment]
@monojenkins commit apidiff

**Comment by marek-safar** [comment]
@monojenkins commit apidiff

**Comment by akoeplinger** [comment]
@monojenkins commit apidiff and csproj

**Pull request merged**